mkdocs/docs/HPC/jupyter.md

```diff
@@ -0,0 +1,138 @@
+
+### Launching a notebook using the web portal
+
+Through the [HPC-UGent web portal](web_portal.md) you can easily start a Jupyter notebook on a workernode, via the *Jupyter Notebook* button under the *Interactive Apps* menu item.
+
+<center>
+![image](img/ood_start_jupyter.png)
+</center>
+
+After starting the Jupyter notebook using the *Launch* button, you will see it being added in state *Queued* in the overview of interactive sessions (see *My Interactive Sessions* menu item):
+
+<center>
+![image](img/ood_jupyter_queued.png)
+</center>
+
+When your job hosting the Jupyter notebook starts running, the status will first change the *Starting*:
+
+<center>
+![image](img/ood_jupyter_starting.png)
+</center>
+
+and eventually the status will change to *Running*, and you will be able to connect to the Jupyter environment using the blue *Connect to Jupyter* button:
+
+<center>
+![image](img/ood_jupyter_running.png)
+</center>
+
+This will launch the Jupyter environment in a new browser tab, where you can open an existing notebook by navigating to the directory where it is located and clicking it. You can also create a new notebook by clicking on `File`>`New`>`Notebook`:
+
+<center>
+![image](img/ood_jupyter_new_notebook.png)
+</center>
+
+### Using extra Python packages
+
+A number of Python packages are readily available in modules on the HPC. To illustrate how to use them in a Jupyter notebook, we will make use of an example where we want to use numpy in our notebook.
+The first thing we need to do is finding the modules that contain our package of choice. For numpy, this would be the SciPy modules.
+
+To find the appropriate modules, it is recommended to use the shell within the web portal under `Clusters`>`>_login Shell Access`.
+
+<center>
+![image](img/ood_jupyter_open_shell.png)
+</center>
+
+We can see all available versions of the SciPy module by using `module available SciPy`:
+
+```shell 
+$ module available Scipy
+
+------------------ /apps/gent/RHEL8/zen2-ib/modules/all ------------------
+    SciPy-bundle/2022.05-foss-2022a    SciPy-bundle/2023.11-gfbf-2023b (D)
+    SciPy-bundle/2023.07-gfbf-2023a
+
+  Where:
+   D:  Default Module
+...
+```
+
+Not all modules will work for every notebook, we need to use the one that uses the same toolchain as the notebook we want to launch. To find that toolchain, we can look at the `JupyterNotebook version` field when creating a notebook. In our example `7.2.0` is the version of the notebook and `GCCcore 13.2.0` is the toolchain used.
+
+<center>
+![image](img/ood_jupyter_version.png)
+</center>
+
+Most modules' names end with the version of the toolchain used by the module (for example `plotly.py/5.18.0-GCCcore-13.2.0`), however for our SciPy module, that is not the case.
+To find the toolchain used by such a module (and the packages contained within a module), we can make use of `module show <module_name>`:
+
+```shell
+$ module show SciPy-bundle/2023.11-gfbf-2023b
```

Comment:
Most of this output is irrelevant, we should also use `grep` here to filter this down a bit?

And you actually want to use `module show gfbf/2023b` here, since only then you're guaranteed to see which `GCCcore` is used as subtoolchain

Reply:
I'm not sure what I should do with this section. Since the toolchain can be found when looking at the module name, what  should people use `module show` for?

Reply:
When using `module show gfbf/2023b`, `GCCcore` isn't mentioned, but `GCC` is, I put `module show gfbf/2023b | grep GCC` in the next commit for now (another option, if you wanted to keep the mention of `GCCcore`, would be `module show SciPy-bundle/2023.11-gfbf-2023b | grep GCCcore`):

```shell
$ module show gfbf/2023b
----------------------------------------------------------------------------------------------------------------------------------
   /apps/gent/RHEL8/zen2-ib/modules/all/gfbf/2023b.lua:
----------------------------------------------------------------------------------------------------------------------------------
help([[
Description
===========
GNU Compiler Collection (GCC) based compiler toolchain, including
 FlexiBLAS (BLAS and LAPACK support) and (serial) FFTW.


More information
================
 - Homepage: (none)
]])
whatis("Description: GNU Compiler Collection (GCC) based compiler toolchain, including
 FlexiBLAS (BLAS and LAPACK support) and (serial) FFTW.")
whatis("Homepage: (none)")
whatis("URL: (none)")
conflict("gfbf")
load("GCC/13.2.0")
load("FlexiBLAS/3.3.1-GCC-13.2.0")
load("FFTW/3.3.10-GCC-13.2.0")
setenv("EBROOTGFBF","/apps/gent/RHEL8/zen2-ib/software/gfbf/2023b")
setenv("EBVERSIONGFBF","2023b")
setenv("EBDEVELGFBF","/apps/gent/RHEL8/zen2-ib/software/gfbf/2023b/easybuild/gfbf-2023b-easybuild-devel")
```

It also might be worth mentioning that, after loading the module, the output for `module show gfbf/2023b` (and any other loaded module) is different and doesn't mention any toolchain:

 ```shell
 $ module show gfbf/2023b
----------------------------------------------------------------------------------------------------------------------------------
   /apps/gent/RHEL8/zen2-ib/modules/all/gfbf/2023b.lua:
----------------------------------------------------------------------------------------------------------------------------------
help([[
Description
===========
GNU Compiler Collection (GCC) based compiler toolchain, including
 FlexiBLAS (BLAS and LAPACK support) and (serial) FFTW.


More information
================
 - Homepage: (none)
]])
whatis("Description: GNU Compiler Collection (GCC) based compiler toolchain, including
 FlexiBLAS (BLAS and LAPACK support) and (serial) FFTW.")
whatis("Homepage: (none)")
whatis("URL: (none)")
conflict("gfbf")
setenv("EBROOTGFBF","/apps/gent/RHEL8/zen2-ib/software/gfbf/2023b")
setenv("EBVERSIONGFBF","2023b")
setenv("EBDEVELGFBF","/apps/gent/RHEL8/zen2-ib/software/gfbf/2023b/easybuild/gfbf-2023b-easybuild-devel")
```

Should this be mentioned in the jupyter page? Or is this too specific (I have included it for now)?

Reply:
Text is OK now, we need a better mechanism to let people figure out what's compatible with what anyway, it's quite hard to explain (but what's there is sort of OK I think)